www/docs/server/error-formatting.md

```diff
@@ -51,6 +51,40 @@ export function MyComponent() {
 }
 ```
 
+## Usage example-prisma conflict
+
+In this one we even change http status code of the response from 500 to 409
+
+### Serverside
+
+```ts title='server.ts'
+import { initTRPC } from '@trpc/server';
+import { Prisma } from '@prisma/client'
+
+export const t = initTRPC.context<Context>().create({
+  errorFormatter(opts) {
+    const { shape, error } = opts;
+      const prismaError = error.cause as Prisma.PrismaClientKnownRequestError;
+      if (prismaError && prismaError.code === 'P2002' && prismaError.meta) {
```

Comment:
nit: But maybe this should be a type-guard helper (unless they export one internally?) ?

```ts
const isPrismaError(cause: unknown): cause is Prisma.PrismaClientKnownRequestError {
  return !!(
    cause &&
    typeof cause === "object" &&
    "code" in cause &&
    typeof cause.code === "string" &&
    cause.code.startsWith("P")
  );
}

// ...

if (isPrismaError(error.cause)) {
  ...
}
```

Reply:
Turns out `Prisma.PrismaClientKnownRequestError` is a class so you can do `instanceof` on it :)